browser/ui/brave_ads/ad_notification_control_buttons_view.cc

```diff
@@ -0,0 +1,112 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/brave_ads/ad_notification_control_buttons_view.h"
+
+#include <memory>
+
+#include "base/bind.h"
+#include "brave/app/vector_icons/vector_icons.h"
+#include "brave/browser/ui/brave_ads/ad_notification_view.h"
+#include "brave/browser/ui/brave_ads/padded_image_button.h"
+#include "brave/browser/ui/brave_ads/padded_image_view.h"
+#include "ui/compositor/layer.h"
+#include "ui/gfx/color_palette.h"
+#include "ui/gfx/paint_vector_icon.h"
+#include "ui/views/background.h"
+#include "ui/views/layout/box_layout.h"
+#include "ui/views/metadata/metadata_impl_macros.h"
+
+namespace brave_ads {
+
+namespace {
+
+const int kMinimumButtonHeight = 44;
+
+const int kInfoButtonIconDipSize = 32;
+
+constexpr SkColor kCloseButtonIconColor = SkColorSetRGB(0x4c, 0x36, 0xd2);
+const int kCloseButtonIconDipSize = 18;
+
+}  // namespace
+
+AdNotificationControlButtonsView::AdNotificationControlButtonsView(
+    AdNotificationView* ad_notification_view)
+    : ad_notification_view_(ad_notification_view) {
+  DCHECK(ad_notification_view_);
+
+  CreateView();
+}
+
+AdNotificationControlButtonsView::~AdNotificationControlButtonsView() = default;
+
+void AdNotificationControlButtonsView::ShowInfoButton(const bool should_show) {
+  if (should_show && !info_button_) {
```

Comment:
I think we can control button visibility by `info_button_->SetVisible()`. Is there a reason it should be created every time?

Reply:
Fixed as these buttons will always be visible. Thanks